File: sujeong/kakao2021/합승 택시 요금.swift
import Foundation

var adj:[[Int]] = [[Int]]()
let MAX_FARE:Int = 20100000

func solution(_ n:Int, _ s:Int, _ a:Int, _ b:Int, _ fares:[[Int]]) -> Int {
    
    initAdjMatrix(n, fares)
    
    floyd(n)
        
    let result = getMinFare(n,s,a,b)
    
    return result
}

func initAdjMatrix(_ n:Int, _ fares:[[Int]]) {
    adj = Array(repeating:Array(repeating:MAX_FARE, count:n+1), count:n+1)
    for i in 1...n {
        adj[i][i] = 0
    }
    for edge in fares {
        let nodeS = edge[0]
        let nodeE = edge[1]
        let fare = edge[2]
        adj[nodeS][nodeE] = fare
        adj[nodeE][nodeS] = fare
    }
}

func floyd(_ n:Int){
    for i in 1...n {
        for r in 1...n {
            for c in 1...n {
                adj[r][c] = min(adj[r][c], adj[r][i] + adj[i][c])
            }
        }
    }
}

func getMinFare(_ n:Int, _ s:Int, _ a:Int, _ b:Int) -> Int{
    var minFare:Int = MAX_FARE
    for m in 1...n {
        minFare = min(minFare, adj[s][m] + adj[m][a] + adj[m][b])
    }
    return minFare
}

File: hyein/kakao2021/합승택시요금.py
from math import inf

def solution(n, s, a, b, fares):
    s, a, b = s - 1, a - 1, b - 1

    graph = [[inf] * n for _ in range(n)]
    for i in range(n):
        graph[i][i] = 0     

    for fare in fares:
        u, v, w = fare
        graph[u - 1][v - 1] = graph[v - 1][u - 1] = w

    for k in range(n):
        for i in range(n):
            for j in range(n):
                if graph[i][j] > graph[i][k] + graph[k][j]:
                    graph[i][j] = graph[i][k] + graph[k][j]
                # graph[i][j] = min(graph[i][j], graph[i][k] + graph[k][j])

    answer = inf
    for k in range(n):
        answer = min(answer, graph[s][k] + graph[k][a] + graph[k][b])

    return answer